cogs/statistics.py

```diff
@@ -12,6 +14,136 @@ def __init__(self, bot):
+            description="Total command statistics")
+        data = await self.generate_embed(ctx, data, cursor)
+        try:
+            await ctx.send(embed=data)
+        except discord.Forbidden:
+            await ctx.send("Need permission to embed links")
+
+
+    async def get_commands(self, cursor, search):
+        """Returns ordered dict of commands from cursor and search string in DB"""
+        commands = {}
+        async for stat in cursor:
+            if stat[search] in commands:
+                commands[stat[search]] += 1
+            else:
+                commands[stat[search]] = 1
+        ordered_commands = collections.OrderedDict(
+            sorted(commands.items(), key=lambda x: x[1], reverse=True))
+        return ordered_commands
+
+    def calc_percentage(self, ordered_commands, total):
+        """Generates ordered dict of percentages of used commands from ordered_commands"""
+        percentages = {}
+        for k, v in ordered_commands.items():
+            percentages[k] = round(100 / total * v)
+        ordered_percentages = collections.OrderedDict(
+            sorted(percentages.items(), key=lambda x: x[1], reverse=True))
+        return ordered_percentages
+
+    async def generate_embed(self, ctx, data, cursor, *, rank=True):
+        # Get data
+        total_amount = await cursor.count()
+        ordered_commands = await self.get_commands(cursor, 'command')
+        percentages = self.calc_percentage(ordered_commands, total_amount)
+        data.add_field(
+            name="Total commands", value=str(total_amount), inline=False)
+        output = self.generate_commands(ordered_commands)
+        data.add_field(name="Most used commands", value=output, inline=False)
+        output = self.generate_diagram(percentages)
+        data.add_field(name="Diagram", value=output, inline=False)
+        if rank:
+            cursor = cursor.rewind()
+            ranking = await self.get_commands(cursor, 'author')
+            output = await self.generate_ranking(ctx, ranking)
+            data.add_field(name="Ranking", value=output, inline=False)
+        return data
+
+    def generate_commands(self, ordered_commands):
+        """Returns the 10 most used commands from ordered_commands"""
+        output = ""
+        counter = 0
+        for k, v in ordered_commands.items():
+            if counter < 10:
+                output += "{0} used {1} times\n".format(k.capitalize(), v)
+                counter += 1
+        return output
+
+    def generate_diagram(self, percentages):
+        """Generates string of ASCII bar out of ordered_dict of percentages"""
+        counter = 0
+        output = ""
+        for k, v in percentages.items():
+            if counter < 5:
+                bar_count = round(v / 5)
+                for emoji in range(0, bar_count):
+                    output += "▓"
+                tab_count = 20 - bar_count
+                for tab in range(0, tab_count):
```

Comment:
Style: range() doesn't need the initial 0, it defaults to it
```py
range(tab_count)
```

Reply:
Removed 0 from range()